lib/openstax/aws/rds_instance.rb

```diff
@@ -0,0 +1,26 @@
+module OpenStax::Aws
+  class RdsInstance
+
+    attr_reader :raw
+
+    delegate_missing_to :@raw
+
+    def initialize(db_instance_identifier:, region:)
+      @raw = Aws::RDS::DBInstance.new(
+        db_instance_identifier,
+        client: Aws::RDS::Client.new(region: region)
+      )
+    end
+
+    def set_rds_master_password( db_instance_identifier: ,password:)
+      raw.modify_rds_instance({
+        db_instance_identifier: db_instance_identifier,
+        apply_immediately: true, 
+        master_user_password: password
+      })
+  
+    end
```

Comment:
A reasonable spec for this class would be the following (which I have not tested and may contain bugs):

```ruby
require 'spec_helper'

RSpec.describe OpenStax::Aws::RdsInstance do
  let(:instance) { described_class.new(db_instance_identifier: "foo", region: "us-east-1") }

  it "changes the master password" do
    expect(instance.raw).to receive(:modify).with(apply_immediately: true, master_user_password: "bar")
    instance.set_master_password(password: "bar")
  end
end
```

This is using mocking approaches to avoid actually calling the underlying AWS code -- it just checks that the AWS code is called with what we expect.  If you run this code live and see problems you may want to come back here and make this spec more realistic.

Also... you may have noticed that what you're buying for yourself with this new class is being able to say:

```ruby
my_stack.resource("Rds").set_master_password(password: "foo")
```

instead of 

```ruby
my_stack.resource("Rds").modify(apply_immediately: true, master_password: "foo")
```

Where the latter case is what you'd get if your `def resource` `case` statement change returned an instance of the Ruby SDK class instead of your new wrapper class.  Your new code isn't hiding too much complexity, so it is up to you whether you keep it or just use the Ruby SDK class as is.  If you think you'll be adding more wrapper logic in the future (other backup orchestration, etc), then it probably makes sense to keep the wrapper.